Source/ParameterEncoder.swift

```diff
@@ -221,6 +221,25 @@ public final class URLEncodedFormEncoder {
             }
         }
     }
+    
+    /// Configures how spaces are encoded.
+    public enum SpaceEncoding {
+        /// Encodes spaces according to normal percent escaping rules (%20).
+        case percentEscaped
+        /// Encodes spaces as `+`,
+        case plusReplaced
+        
+        /// Encodes the string according to the encoding.
+        ///
+        /// - Parameter string: The `String` to encode.
+        /// - Returns:          The encoded `String`.
+        func encode(_ string: String) -> String {
+            switch self {
+            case .percentEscaped: return string.replacingOccurrences(of: " ", with: "%20")
+            case .plusReplaced: return string.replacingOccurrences(of: " ", with: "+")
```

Comment:
Is the `plusReplaced`  case used to support encoding types for `application/x-www-form-urlencoded `  ?

Reply:
It's actually a commit from the parent of this PR's branch (#2637). I rebranched and it made it part of this PR too.